src/fuzz/README.md

```diff
@@ -0,0 +1,119 @@
+# Fuzzing libspark
+
+## Quickstart Guide
+### Dependencies
+1. Install honggfuzz (https://github.com/google/honggfuzz)
+```
+sudo apt-get install binutils-dev libunwind-dev libblocksruntime-dev clang
+git clone https://github.com/google/honggfuzz.git
+cd honggfuzz
+make
+sudo make install
+```
+
+2. Build firo: 
+
+Follow the instructions from https://github.com/firoorg/firo/tree/spark#readme
+
+### Fuzzing using honggfuzz
+* In order to fuzz `firo/src/libpark` using Honggfuzz:
+
+```
+git clone -b spark_fuzz_blog https://github.com/hashcloak/firo.git
+cd firo/src/fuzz/
+export CC=hfuzz-clang
+export CXX=hfuzz-clang++
+```
+
+To compile with `hfuzz-clang++`:
+
+```
+cd src/fuzz/
+make <filename>
+```
+
+For example(for bpplus):
+```
+cd src/fuzz/
+make bpplus
+```
+The above command will generate an instrumented binary with name `<filename>_hfuzz` (eg: bpplus_hfuzz) inside src/fuzz/libspark.
+The fuzzing harness of the following spark files is availabe: aead, bech32, bpplus, chaum, coin, f4grumble, grootle, mint_transaction, schnorr and spend_transaction.
+
+* To start fuzzing:
+
+1. create a directory to save all the crahses.
+2. Inside the directory run:
+```
+hongfuzz -i <path_of_input_corpora>/<filename_inputs> -- ./<filename_hfuzz> ___FILE___
+```
+
+example: 
+1. `mkdir src/fuzz/bpplus_results && cd src/fuzz/bpplus_results`
+2. `hongfuzz -i ../../inputs/bpplus_inputs -- ./../../libspark/bpplus_hfuzz ___FILE___`
+3. To stop press `ctrl+c`
+
+If there is no input corpora, empty corpora can be provided.
```

Comment:
Please provide a note that this means creating and passing an empty directory via `-i`.